### PR TITLE
Prokka and prodigal modules update

### PR DESCRIPTION
**Pull request by danilodileo**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/metatdenovo/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/metatdenovo _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

I made changes in prodigal output + added a '--metagenome' to prokka. 

as mentioned in prokka's website https://github.com/tseemann/prokka/issues/399 prokka doesn't provide the same output as prodigal, it turns out with less gene predicted. Prokka people might not be interested in fix this issue. As the author wrote on GitHub "Prokka is not really designed for metagenomes."

### CODE REVIEW

File: modules/local/format_prodigal.nf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "conda-forge::gzip=1.12" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gzip:1.11 ':
+        'quay.io/biocontainers/gzip:1.11 ' }"
+
+    input:
+    tuple val(meta), path(eggnog)
```
Comment (erikrikarddaniel) [inline modules/local/format_prodigal.nf]: Why `eggnog`? Confusing.

File: modules/local/format_prodigal.nf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args   ?: ''
+    prefix   = task.ext.prefix ?: "${meta.id}"
+
+    """
+
```
Comment (erikrikarddaniel) [inline modules/local/format_prodigal.nf]: Remove the empty line (I think; i.e. follow conventions).

File: subworkflows/local/prodigal.nf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    take:
+        fastafile
+
+    main:
+        ch_versions = Channel.empty()
+        
+        PRODIGAL_MODULE ( fastafile, 'gff' )
+        FORMAT_PRODIGAL ( PRODIGAL_MODULE.out.gene_annotations )
+
+
```
Comment (erikrikarddaniel) [inline subworkflows/local/prodigal.nf]: Remove.

File: subworkflows/local/prodigal.nf [hunk 4]
```diff
@@ -0,0 +1,28 @@
+//
+// Run prodigal as orf caller then generate nice format for gff
+//
+
+include { PRODIGAL as PRODIGAL_MODULE    } from '../../modules/nf-core/prodigal/main'
+include { FORMAT_PRODIGAL                } from '../../modules/local/format_prodigal.nf'
+include { UNPIGZ as UNPIGZ_FASTA_PROTEIN } from '../../modules/local/unpigz.nf'
```
Comment (erikrikarddaniel) [inline subworkflows/local/prodigal.nf]: You're not using this, I think.

File: workflows/metatdenovo.nf [hunk 5]
```diff
@@ -72,7 +72,6 @@ ch_multiqc_custom_methods_description = params.multiqc_methods_description ? fil
 // MODULE: local
 //
 include { MEGAHIT_INTERLEAVED              } from '../modules/local/megahit/interleaved.nf'
-include { UNPIGZ as UNPIGZ_FASTA_PROTEIN   } from '../modules/local/unpigz.nf'
```
Comment (erikrikarddaniel) [inline workflows/metatdenovo.nf:75]: You moved this to the new subworkflow, but never used it. This makes me wonder which output files from the three orf callers are gzipped and which are not; and do we need to do something about this?

**Review by erikrikarddaniel — COMMENTED** [review]
Some small comments. We should make an inventory of what's gzipped and what's not from the three orf callers.

File: modules/local/format_prodigal.nf (hunk 1 above)
Comment (danilodileo) [reply modules/local/format_prodigal.nf]: I used eggnog_table.nf as template but I forgot to change the names in some areas :) I changed it.

File: subworkflows/local/prodigal.nf (hunk 4 above)
Comment (danilodileo) [reply subworkflows/local/prodigal.nf]: right. I removed it . I noticed that I didn't need to unpigz while I was working on it but I forgot it there.

File: workflows/metatdenovo.nf (hunk 5 above)
Comment (danilodileo) [reply workflows/metatdenovo.nf:75]: it seems that transdecoder still needs to unpigz

**Pull request merged**